flow/connectors/mysql/cdc.go

```diff
@@ -370,7 +370,10 @@ func (c *MySqlConnector) PullRecords(
 						}
 						return nil
 					}
-					return schema.Columns[idx]
+					if idx < len(schema.Columns) {
+						return schema.Columns[idx]
+					}
+					return nil
```

Comment:
should add a warning here since data read would be incomplete, but could spam so try to hushWarn